isthmus/src/main/java/io/substrait/isthmus/expression/ExpressionRexConverter.java

```diff
@@ -0,0 +1,159 @@
+package io.substrait.isthmus.expression;
+
+import io.substrait.expression.AbstractExpressionVisitor;
+import io.substrait.expression.Expression;
+import io.substrait.expression.FieldReference;
+import io.substrait.isthmus.TypeConverter;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import org.apache.calcite.avatica.util.ByteString;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.util.TimeString;
+import org.apache.calcite.util.TimestampString;
+
+public class ExpressionRexConverter extends AbstractExpressionVisitor<RexNode, RuntimeException> {
+  private final RelDataTypeFactory typeFactory;
+  private final RexBuilder rexBuilder;
+  private final ScalarFunctionConverter scalarFunctionConverter;
+
+  public ExpressionRexConverter(
+      RelDataTypeFactory typeFactory, ScalarFunctionConverter scalarFunctionConverter) {
+    this.typeFactory = typeFactory;
+    this.rexBuilder = new RexBuilder(typeFactory);
+    this.scalarFunctionConverter = scalarFunctionConverter;
+  }
+
+  @Override
+  public RexNode visit(Expression.NullLiteral expr) throws RuntimeException {
+    return null;
```

Comment:
This looks wrong. I think you should be making a typed null literal here, no?

Reply:
Changed to return a typed null literal.  Also changed ut to cover the substrait -> rex for typed null case.